src/trajectory.jl

```diff
@@ -641,21 +641,29 @@ find_good_eps(
     max_n_iters::Int=100
 ) where {T<:AbstractFloat} = find_good_eps(GLOBAL_RNG, h, θ; max_n_iters=max_n_iters)
 
-_rand(rng::AbstractRNG) = rand(rng)
-_rand(rng::AbstractVector{<:AbstractRNG}) = rand.(rng)
-
 """
 Perform MH acceptance based on energy, i.e. negative log probability.
 """
 function mh_accept_ratio(
-    rng::Union{AbstractRNG,AbstractVector{<:AbstractRNG}},
-    Horiginal::AbstractScalarOrVec{<:T},
-    Hproposal::AbstractScalarOrVec{<:T}
+    rng::AbstractRNG,
+    Horiginal::T,
+    Hproposal::T,
```

Comment:
This looks good to me. 

@xukai92  I think the old implementation has some type ambiguity, because it does not guarantee variables `rnd`, `Horiginal` and `Hproposal` to be scalar or vector at the same time. The type inconsistency among these variables might be the cause of the compiler error (JuliaLang/julia#34232).

Reply:
I think it's more subtle than that. For one, the error didn't show up before when DynamicPPL was not a thing. Also, dispatch is not entirely the issue as I show in `mh2` in JuliaLang/julia#34232.

Reply:
Also, the function actually runs correctly even with the compiler errors there, and no run-time ambiguity error is thrown, so...it's most likely not an everyday ambiguity error.

Reply:
Btw, this error is only on Julia 1.0 and 1.1. Julia 1.2 and 1.3 work fine.